api/src/main/java/org/apache/iceberg/catalog/Catalog.java

```diff
@@ -388,6 +392,59 @@ default TableBuilder buildTable(TableIdentifier identifier, Schema schema) {
   default void initialize(String name, Map<String, String> properties) {
   }
 
+  /**
+   * Get catalog properties.
+   *
+   * @return catalog properties
+   */
+  Map<String, String> properties();
+
+  /**
+   * Get default table properties set at Catalog level through catalog properties.
+   *
+   * @return default table properties specified in catalog properties
+   */
+  default Map<String, String> tableCreateDefaultProperties() {
+    String tableCreateDefaultPrefix = "table.create.default.";
+    Map<String, String> props = properties() == null ? Collections.emptyMap() : properties();
+
+    return props.entrySet().stream()
+        .filter(e -> e.getKey().toLowerCase(Locale.ROOT).startsWith(tableCreateDefaultPrefix))
+        .collect(Collectors.toMap(e -> e.getKey().replace(tableCreateDefaultPrefix, ""),
+            Map.Entry::getValue));
+  }
+
+  /**
+   * Get table properties that are enforced at Catalog level through catalog properties.
+   *
+   * @return default table properties enforced through catalog properties
+   */
+  default Map<String, String> tableCreateEnforcedProperties() {
+    String tableCreateEnforcedPrefix = "table.create.enforced.";
+    Map<String, String> props = properties() == null ? Collections.emptyMap() : properties();
+
+    return props.entrySet().stream()
+        .filter(e -> e.getKey().toLowerCase(Locale.ROOT).startsWith(tableCreateEnforcedPrefix))
+        .collect(Collectors.toMap(e -> e.getKey().replace(tableCreateEnforcedPrefix, ""),
+            Map.Entry::getValue));
+  }
+
+  /**
+   * Return updated table properties with table properties defaults and enforcements set at Catalog level through
+   * catalog properties.
+   *
+   * @return updated table properties with defaults and enforcements set at Catalog level
+   */
+  default Map<String, String> overrideTableProperties(Map<String, String> tableProperties) {
+    Map<String, String> props = tableProperties == null ? Collections.emptyMap() : tableProperties;
+
+    return Stream.concat(Stream.concat(
+        tableCreateDefaultProperties().entrySet().stream(),
+            props.entrySet().stream()),
```

Comment:
This mixes indentation for semantic levels. See https://github.com/apache/iceberg/blob/master/CONTRIBUTING.md#line-breaks

Reply:
Thanks for the pointer, updated.